library/pkcs7.c

```diff
@@ -481,11 +489,16 @@ static int pkcs7_get_signed_data(unsigned char *buf, size_t buflen,
     }
 
     /* Do not expect any content */
-    ret = pkcs7_get_content_info_type(&p, end_set, &signed_data->content.oid);
+    ret = pkcs7_get_content_info_type(&p, end_set, &end_content_info,
+                                      &signed_data->content.oid);
     if (ret != 0) {
         return ret;
```

Comment:
I think the issue is that here, we need to distinguish between valid content present (which is unsupported), vs invalid content (invalid data). Something like the following would work:

```suggestion
        /* Determine if valid content is present */
        ret = mbedtls_asn1_get_tag(&p, end_content_info, &len, MBEDTLS_ASN1_CONSTRUCTED | MBEDTLS_ASN1_CONTEXT_SPECIFIC);
        if (ret != 0) {
            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_PKCS7_INVALID_CONTENT_INFO, ret);
        }
        ret = mbedtls_asn1_get_tag(&p, end_content_info, &len, MBEDTLS_ASN1_OCTET_STRING);
        if (ret != 0) {
            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_PKCS7_INVALID_CONTENT_INFO, ret);
        }
        p += len;
        if (p != end_content_info) {
            return MBEDTLS_ERROR_ADD(MBEDTLS_ERR_PKCS7_INVALID_CONTENT_INFO, ret);
        }
        /* Valid content is present - this is not supported */
        return MBEDTLS_ERR_PKCS7_FEATURE_UNAVAILABLE;
```

(this would mean reverting the changes to the expected result for the two `pkcs7_get_signers_info_set error handling` cases, which I think makes sense).

The only downside is that this introduces a couple of uncovered lines - but as they are returning errors in a block that can only return an error, I'm OK with this.

Reply:
@tom-cosgrove-arm WDYT?

Reply:
> The only downside is that this introduces a couple of uncovered lines - but as they are returning errors in a block that can only return an error, I'm OK with this.

I would still prefer to see tests for them, especially because future changes may add support for embedded data.

Reply:
Actually, in this case, let's get the code in and work on those tests later, please